src/partial-result-stream.ts

```diff
@@ -355,13 +359,25 @@ export function partialResultStream(
     },
   });
 
+  // This listener ensures that the last request that executed successfully
+  // after one or more retries will end the requestsStream.
+  const endListener = () => {
+    if (lastErr) {
+      setImmediate(() => requestsStream.end());
+    }
+  };
   const makeRequest = (): void => {
-    requestsStream.add(requestFn(lastResumeToken));
+    if (lastRequestStream) {
+      lastRequestStream.removeListener('end', endListener);
+    }
+    lastRequestStream = requestFn(lastResumeToken);
```

Comment:
I wonder if a retryable error happens at an initial call, `lastResumeToken` is empty here, so what would `lastRequestStream` be here? In other words, what will `requestFn` return for an empty `lastResumeToken`?

Reply:
This code is also used for the initial call to `requestStreamingSql`, which is also always called without a resume token. Calling `requestFn` another time without a resume token will start a new stream. So:
* The first time this is called (the initial call), `lastRequestStream` is `undefined` at first (i.e. the condition of the `if` statement will be `false`).
* If the initial call fails with a retryable error, this method is called again with `lastRequestStream` still pointing to the initial stream. The `endListener` will then be removed from the initial call and placed on the second call.